File: golang-for-nodejs-developers/examples/url_parse.js
const urlstr = 'http://bob:secret@sub.example.com:8080/somepath?foo=bar';

const parsed = new URL(urlstr);
console.log(parsed.protocol);
console.log(`${parsed.username}:${parsed.password}`);
console.log(parsed.port);
console.log(parsed.hostname);
console.log(parsed.pathname);

const query = Object.fromEntries(parsed.searchParams);
console.log(query);

File: golang-for-nodejs-developers/examples/url_parse.go
package main

import (
	"fmt"
	"net/url"
)

func main() {
	urlstr := "http://bob:secret@sub.example.com:8080/somepath?foo=bar"

	u, err := url.Parse(urlstr)
	if err != nil {
		panic(err)
	}

	fmt.Println(u.Scheme + ":") // To match JS output if protocol includes colon
	fmt.Println(u.User)
	fmt.Println(u.Port())
	fmt.Println(u.Hostname())
	fmt.Println(u.Path)
	fmt.Println(u.Query())
}
